explainer.md

```diff
@@ -415,17 +415,21 @@ xrSession.addEventListener('inputsourceschange', (ev) => {
 });
 ```
 
+The properties of an XRInputSource object are immutable. If a device can be manipulated in such a way that these properties can change, the `XRInputSource` will be removed and recreated.
```

Comment:
I had to think about it for a while to decide if I'm OK with this, but I think I am. The most problematic scenario that I can think of is if you have a system that is not explicitly handed (like the Vive wands) and the system decides that you've switched hands in the middle of a grab-and-drag gesture it would be common for the page implementation to cause the object to be dropped as the "hand" holding it disappears.

The reality is, though, that I rarely see this happen outside of early system initialization and if it's a large concern then the system can silently re-map the physical devices to the previously created left/right hand input sources to hide the transition. Plus there's a significant number of devices for which the handedness is an explicit property of the input device or only changed manually, and therefore will never be surprising to the user.

So TL;DR: I'm good with it!

Reply:
It is precisely since this is rare that we believe it is safer to remove and re-add the sources - since most apps will experience effectively unchanging sources, they'll likely assume they don't change any would probably miss such changes.  Removing and re-adding the source gets us back onto the mainline path such apps will have tested.